plugins/robots/generators/nxt/nxtOsekCGenerator/nxtFlashTool.cpp

```diff
@@ -222,15 +223,26 @@ bool NxtFlashTool::uploadProgram(const QFileInfo &fileInfo)
 	mCompileState = idle;
 	mSource = fileInfo;
 
-#ifdef Q_OS_WIN
 	mCompileProcess.setWorkingDirectory(path());
-	mCompileProcess.start("cmd", { "/c", path("compile.bat")
-			+ " " + fileInfo.completeBaseName()
-			+ " " + fileInfo.absolutePath() });
-#else
-	mCompileProcess.start("sh", { path("compile.sh") , fileInfo.absolutePath()});
-#endif
 
+	switch(QOperatingSystemVersion::currentType()) {
+	default:
+		break;
+	case QOperatingSystemVersion::OSType::Windows: {
+		auto pathToNxtTools = path().replace("\\", "/");
+		auto line = path("compile.bat")
+					+ " " + fileInfo.completeBaseName()
```

Comment:
Probably, file names with spaces are handled incorrectly here after the single line is combined.